integration/container/attach_test.go

```diff
@@ -0,0 +1,206 @@
+// +build !windows
+
+package container
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+	"gotest.tools/poll"
+	"gotest.tools/skip"
+)
+
+// #9860 Make sure attach ends when container ends (with no errors)
+func TestAttachClosedOnContainerStop(t *testing.T) {
+	skip.If(t, testEnv.IsRemoteDaemon, "test requires daemon on the same host")
+
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(t, ctx, client, container.WithTty(true),
+		container.WithCmd("/bin/sh", "-c", `trap 'exit 0' SIGTERM; while true; do sleep 1; done`),
+		func(c *container.TestContainerConfig) {
```

Comment:
Could you can add a `container.WithStdIn(bool)` function for this in https://github.com/moby/moby/blob/master/integration/internal/container/ops.go ?

Reply:
Thanks for the review, done.